src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1216,13 +1222,13 @@ def aks_agentpool_upgrade(cmd,
     )
     if kubernetes_version != '' and node_image_only:
         raise MutuallyExclusiveArgumentError(
-            'Conflicting flags. Upgrading the Kubernetes version will also '
-            'upgrade node image version. If you only want to upgrade the '
-            'node version please use the "--node-image-only" option only.'
+            'Conflicting flags. Unable to specify max-surge/drain-timeout/node-soak-duration with node-image-only.'
+            'If you want to use max-surge/drain-timeout/node-soak-duration with a node image upgrade, please first '
+            'update max-surge/drain-timeout/node-soak-duration using "az aks nodepool update --max-surge/--drain-timeout/--node-soak-duration".'
```

Comment:
This error message should be the description of the exception on line 1232?

Reply:
Indeed! [Fixed](https://github.com/Azure/azure-cli-extensions/pull/6865/commits/299794bf4d95e280b9acb0cd5ffde679dde968c8)